cmd/manager/main.go

```diff
@@ -29,34 +29,59 @@ type server struct {
 }
 
 func (s *server) CreateStudy(ctx context.Context, in *api_pb.CreateStudyRequest) (*api_pb.CreateStudyReply, error) {
+	var studyID string
+	var err error
 	if in == nil || in.StudyConfig == nil {
 		return &api_pb.CreateStudyReply{}, errors.New("StudyConfig is missing.")
 	}
-	studyID, err := dbIf.CreateStudy(in.StudyConfig)
-	if err != nil {
-		return &api_pb.CreateStudyReply{}, err
+
+	if in.StudyConfig.JobType != "nas" {
+		//If it is a HP job
+		studyID, err = dbIf.CreateStudy(in.StudyConfig)
+		if err != nil {
+			return &api_pb.CreateStudyReply{}, err
+		}
+		s.SaveStudy(ctx, &api_pb.SaveStudyRequest{
+			StudyName:   in.StudyConfig.Name,
+			Owner:       in.StudyConfig.Owner,
+			Description: "StudyID: " + studyID,
+		})
+
+	} else {
+		//If it is a NAS job
+		studyID = "1"
+		log.Printf("NAS job is created, id= %v", studyID)
 	}
-	s.SaveStudy(ctx, &api_pb.SaveStudyRequest{
-		StudyName:   in.StudyConfig.Name,
-		Owner:       in.StudyConfig.Owner,
-		Description: "StudyID: " + studyID,
-	})
 	return &api_pb.CreateStudyReply{StudyId: studyID}, nil
 }
 
 func (s *server) DeleteStudy(ctx context.Context, in *api_pb.DeleteStudyRequest) (*api_pb.DeleteStudyReply, error) {
 	if in == nil || in.StudyId == "" {
 		return &api_pb.DeleteStudyReply{}, errors.New("StudyId is missing.")
 	}
-	err := dbIf.DeleteStudy(in.StudyId)
-	if err != nil {
-		return &api_pb.DeleteStudyReply{}, err
+	if in.JobType != "nas" {
+		//If it is a HP job
+		err := dbIf.DeleteStudy(in.StudyId)
+		if err != nil {
+			return &api_pb.DeleteStudyReply{}, err
+		}
+	} else {
+		//If it is a NAS job
+		log.Printf("NAS job is deleted, id= %v", in.StudyId)
```

Comment:
We do not need to delete the study, could you please give me more details here?

Reply:
See the commentary :)